### PR TITLE
upgrading php and symfony

### PR DESCRIPTION
**Pull request by atarisafari**
I tested this to make sure everything compiles but make sure to do your own testing as well.

Hopefully this will solve the sessionAuthenticator issue you all have been having!

### CODE REVIEW

**Comment by rob-3** [comment]
Does this require the docker PR as well? `docker-compose up --build` errors out with `FileNotFoundError: [Errno 2] No usable temporary directory found in ['/tmp', '/var/tmp', '/usr/tmp', '/var/www/html']` for me.

**Comment by sgcooper78** [comment]
@rob-3 I believe the first error has to do with your file system. I was able to pull down the commit and run docker compose build and bring it up without errors for this branch. 

Also for the other error we don't use PostgreSQL at Cidilabs but you have to configure it, I will add it to the docker branch.
```
RUN apt-get install -y libpq-dev \
    && docker-php-ext-configure pgsql -with-pgsql=/usr/local/pgsql \
    && docker-php-ext-install pdo pdo_pgsql pgsql
```

**Comment by rob-3** [comment]
What configuration do you have set in `.env.local`? Could that be causing my build to fail?

I think the first time was an memory issues because I'm on a limited AWS instance, but this time the build finished and I got an error in the startup:
```
db_1   | 2022-04-22T14:36:44.160401Z 1 [ERROR] [MY-012224] [InnoDB] Tablespace flags are invalid in datafile: ./ibdata1, Space ID:0, Flags: 21. Please refer to http://dev.mysql.com/doc/refman/8.0/en/innodb-troubleshooting-datadict.html for how to resolve the issue.
db_1   | 2022-04-22T14:36:44.161073Z 1 [ERROR] [MY-012237] [InnoDB] Corrupted page [page id: space=0, page number=0] of datafile './ibdata1' could not be found in the doublewrite buffer.
db_1   | 2022-04-22T14:36:44.162959Z 1 [ERROR] [MY-012930] [InnoDB] Plugin initialization aborted with error Data structure corruption.
db_1   | 2022-04-22T14:36:44.653857Z 1 [ERROR] [MY-011013] [Server] Failed to initialize DD Storage Engine.
db_1   | 2022-04-22T14:36:44.654766Z 0 [ERROR] [MY-010020] [Server] Data Dictionary initialization failed.
db_1   | 2022-04-22T14:36:44.655254Z 0 [ERROR] [MY-010119] [Server] Aborting
db_1   | 2022-04-22T14:36:44.656191Z 0 [System] [MY-010910] [Server] /usr/sbin/mysqld: Shutdown complete (mysqld 8.0.28)  MySQL Community Server - GPL.
```

**Comment by rob-3** [comment]
Also, anyone with an M1 Mac or other arm system can't build this because `mysql` doesn't publish arm images to Docker. One workaround is to swap `mariadb` for `mysql` in `docker-compose.yml`

**Comment by sgcooper78** [comment]
I included the changes in the other PR.  We use the .env.local.example in that branch.
You shouldn't be getting any errors on bringing up the DB. I switched the DB to use the file-system instead of ephemeral for docker-compose. try deleting the database directory if you have one and then bringing it back up. It should be a fresh version after you delete it. 

We also found this issue out about mysql and M1's and switched a couple weeks ago.

**Comment by rob-3** [comment]
> We also found this issue out about mysql and M1's and switched a couple weeks ago.

Wait - should that change be in this PR? We still have MySQL in our repo.

File: src/Security/SessionAuthenticator.php [hunk 1]
```diff
@@ -4,76 +4,86 @@
...
     }
 
-    public function onAuthenticationSuccess(Request $request, TokenInterface $token, $providerKey)
+    public function onAuthenticationFailure(Request $request, AuthenticationException $exception): ?Response
     {
         return null;
     }
 
-    public function onAuthenticationFailure(Request $request, AuthenticationException $exception) 
+    public function start(Request $request, AuthenticationException $exception = null)
```
Comment (rob-3) [inline src/Security/SessionAuthenticator.php:60]: Is there any reason to keep `start()` around? I took a look at the new interface and it doesn't seem to have this method anymore.

**Comment by sgcooper78** [comment]
@rob-3 Yes, I would move to MariaDB in the docker-compose files. I have updated the PR.

**Comment by rob-3** [comment]
This is just waiting on a resolution of whether we can delete `start()` entirely. There are also some unused imports and fields that would ideally be removed as well before this is merged.

**Pull request merged**